webapp/channels/src/components/admin_console/filter/filter_checkbox.tsx

```diff
@@ -10,44 +10,42 @@ type Props = {
     updateOption: (checked: boolean, name: string) => void;
 };
 
-class FilterCheckbox extends React.PureComponent<Props> {
-    toggleOption = (e: React.MouseEvent) => {
+function FilterCheckbox(props: Props) {
+    const {name, checked, label} = props;
+
+    function toggleOption(e: React.MouseEvent) {
         e.preventDefault();
         e.stopPropagation();
-        const {checked, name, updateOption} = this.props;
+        const {checked, name, updateOption} = props;
         updateOption(!checked, name);
-    };
-
-    render() {
-        const {name, checked, label} = this.props;
-        return (
-            <div
-                className='FilterList_checkbox'
-                onClick={this.toggleOption}
-            >
-                <label>
-                    {checked &&
-                        <input
-                            type='checkbox'
-                            id={name}
-                            name={name}
-                            defaultChecked={true}
-                        />
-                    }
-
-                    {!checked &&
-                        <input
-                            type='checkbox'
-                            id={name}
-                            name={name}
-                            defaultChecked={false}
-                        />
-                    }
-                    {label}
-                </label>
-            </div>
-        );
     }
+    return (
+        <div
+            className='FilterList_checkbox'
+            onClick={toggleOption}
+        >
+            <label>
+                {checked &&
+                    <input
+                        type='checkbox'
+                        id={name}
+                        name={name}
+                        defaultChecked={true}
+                    />
+                }
+
+                {!checked &&
+                    <input
+                        type='checkbox'
+                        id={name}
+                        name={name}
+                        defaultChecked={false}
+                    />
+                }
```

Comment:
```suggestion
                    <input
                        type='checkbox'
                        id={name}
                        name={name}
                        defaultChecked={checked}
                    />
```

What do you think @larkox?

Reply:
I agree at least. I believe I just did not want to change anything other than Class Component to Function Component and thus left it unchanged.

Reply:
You are completely right. It is out of the scope of the PR, but I would appreciate if you make the improvement @ludvigbolin .

Thanks for pointing it out @Syed-Ali-Abbas-Zaidi

Reply:
I tried running the webapp with this change and it left the checkbox broken for some reason. Do not know why as I agree that it is a reasonable change.

Reply:
I found the issue. This is working for all the wrong reasons :sweat_smile:

So... We are showing the check mark with an input element, which is not what we should do, since it is not a real input element.

If we try to use `checked`, react will complain that if there is not `onChange`, there is no sense to add the checked. And you probably want to use `defaultChecked`. So... probably we made that change at some point about that.

Now the thing is... defaultChecked only change the state on the first render. So changing it in different renders doesn't make any difference. But the way it is implemented right now... they are two different components, not the same component, so it is not a prop changing but the whole component changing. So it "apparently works".

If we want to keep using an input to do this, the "correct" way (if I am not mistaken) is to use `checked` instead of `defaultChecked` but also adding the `readOnly` prop.

Would you take a look at that?

Reply:
I looked into it some more. It almost worked by simply changing it to `checked` instead of `defaultChecked`. But then I could not click on the actual checkbox but instead on the text only. To change this I found a fix by adding a random key. I believe this would re-render it though? Or this it only change the key, thus making it better with this change?

Reply:
@ludvigbolin I am really sorry for all the back and forth. This seems wrong as a base, and fixing it will require more involvement, so let's revert back to how it was before (`{checked && ...}{!checked && ...}`).

Reply:
Alright! No worries. It might be better to keep the PR as it was originally intended.